sjsonnet/src/sjsonnet/PrettyYamlRenderer.scala

```diff
@@ -0,0 +1,434 @@
+package sjsonnet
+
+import java.io.{StringWriter, Writer}
+import java.util.regex.Pattern
+
+import ujson.BaseRenderer
+import upickle.core.{ArrVisitor, ObjVisitor}
+
+/**
+ * A version of YamlRenderer that tries its best to make the output YAML as
+ * pretty as possible: unquoted strings, de-dented lists, etc. Follows the PyYAML
+ * style. Also adds the ability to stream writes to a generic `Writer`.
+ */
+class PrettyYamlRenderer(out: Writer = new java.io.StringWriter(),
+                         indentArrayInObject: Boolean = false,
+                         indent: Int = 2,
```

Comment:
`indent` gets initialized with `config.indent` which defaults to 3. Should the default depend on the output format instead?

Reply:
Good question, we currently explicitly pass in `--indent 2` in our own codebase, so it's a bit of a moot point. In general, we never initialize any of the `FooRenderer`s without passing in an explicit `indent` anywhere in this codebase, so maybe we can just get rid of the default...